### PR TITLE
fix: Correctly resolve the Solution path on non-Windows targets

### PR DESCRIPTION
**Pull request by 96malhar**
*Issue #, if available:*

*Description of changes:*
This PR correctly resolves the Solution path while parsing the project definition on non-Windows targets.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by philasmar — APPROVED** [review]

File: src/AWS.Deploy.Common/ProjectDefinitionParser.cs [hunk 1]
```diff
@@ -126,7 +126,9 @@ private async Task<bool> ValidateProjectInSolution(string projectPath, string so
 
             var lines = await _fileManager.ReadAllLinesAsync(solutionFile);
             var projectLines = lines.Where(x => x.StartsWith("Project"));
-            var projectPaths = projectLines.Select(x => x.Split(',')[1].Replace('\"', ' ').Trim()).ToList();
+            var projectPaths = projectLines.Select(x => x.Split(',')[1].Replace('\"', ' ').Trim())
+                .Select(x => x.Replace('\\', Path.DirectorySeparatorChar))
+                .ToList();
```
Comment (ganeshnj) [inline src/AWS.Deploy.Common/ProjectDefinitionParser.cs:131]: We should remove this method than fixing  and start using https://github.com/aws/aws-dotnet-deploy/blob/cc7cbcffb353d41aa2c73ead71468d66c80ab72c/src/AWS.Deploy.Orchestration/DeploymentBundleHandler.cs#L195

**Review by ganeshnj — CHANGES_REQUESTED** [review]

File: src/AWS.Deploy.Common/ProjectDefinitionParser.cs [hunk 2]
```diff
@@ -126,7 +126,9 @@ private async Task<bool> ValidateProjectInSolution(string projectPath, string so
 
             var lines = await _fileManager.ReadAllLinesAsync(solutionFile);
             var projectLines = lines.Where(x => x.StartsWith("Project"));
-            var projectPaths = projectLines.Select(x => x.Split(',')[1].Replace('\"', ' ').Trim()).ToList();
+            var projectPaths = projectLines.Select(x => x.Split(',')[1].Replace('\"', ' ').Trim())
```
Comment (normj) [inline src/AWS.Deploy.Common/ProjectDefinitionParser.cs:129]: I think we should be defensive an not always assume there is a `1` element. I know that is not new in this PR.

**Review by normj — APPROVED** [review]
Approved but I suspect we are missing some unit tests for this function.

File: src/AWS.Deploy.Common/ProjectDefinitionParser.cs (hunk 1 above)
Comment (96malhar) [reply src/AWS.Deploy.Common/ProjectDefinitionParser.cs]: This method is redundant. So removed it.

**Review by ganeshnj — APPROVED** [review]

**Review by normj — APPROVED** [review]

**Review by philasmar — APPROVED** [review]

**Pull request merged**